### ISSUE TITLE
Parse failure for MT103 example

### ISSUE THREAD

**Issue by danielquinn** [danielquinn/mt103#3]
[Reported](https://gist.github.com/danielquinn/5347d332c4875f2df652265e664454d1) by @bbroto06, the MT103 sample found [here](https://www.paymentcomponents.com/demo/mt/submit) doesn't parse against this library.

Sample string:

```
{1:F01AAAAGRA0AXXX0057000289}{2:O1030919010321BBBBGRA0AXXX00570001710103210920N}{3:{108:MT103 003 OF 045}{121:c8b66b47-2bd9-48fe-be90-93c2096f27d2}}{4:
:20:5387354
:23B:CRED
:23E:PHOB/20.527.19.60
:32A:000526USD1101,50
:33B:USD1121,50
:50K:FRANZ HOLZAPFEL GMBH
VIENNA
:52A:BKAUATWW
:59:723491524
C. KLEIN
BLOEMENGRACHT 15
AMSTERDAM
:71A:SHA
:71F:USD10,
:71F:USD10,
:72:/INS/CHASUS33
-}{5:{MAC:75D138E4}{CHK:DE1B0D71FA96}}
```

**Comment by bbroto06** [comment]
1. "Reported by @bbroto06, the MT103 sample found here doesn't parse against this library."
**here** = **404 Not Found**
you have to put the following link :
https://www.paymentcomponents.com/demo/mt

2. Then click on **MT103** tab

**Comment by danielquinn** [comment]
All fixed!

Thanks for reporting this, 'cause it gave me an excuse to dig a littler further into the regex that underpins this library.  In this case the problem was in a few places: Section 2 of your sample started with an `O` (for outgoing) and I've only ever had to deal with incoming (`I`) ones.  Section 3 had spaces in the MUR, which I'd not seen before either, and section 5 had text that included curly braces `{ }`, and my regex couldn't handle that either.

Anyway I've updated `master` and pushed a patch release to PyPI, so you should be able to `pip install mt103` and it should work now.

I'm going to close this, but if you find there's still problems, feel free to re-open.

**Comment by bbroto06** [comment]
Thanks Daniel for taking time out from your schedule, to fix the issue. Now basic header looks good.
However, the bank op code is still none.
This is the output that I get :
```
 basic header: F01AAAAGRA0AXXX0057000289, 
 bank op code: None, 
 complete message: {1:F01AAAAGRA0AXXX0057000289}{2:O1030919010321BBBBGRA0AXXX00570001710103210920N}{3:{108:MT103 003 OF 045}{121:c8b66b47-2bd9-48fe-be90-93c2096f27d2}}{4:
:20:5387354
:23B:CRED
:23E:PHOB/20.527.19.60
:32A:000526USD1101,50
:33B:USD1121,50
:50K:FRANZ HOLZAPFEL GMBH
VIENNA
:52A:BKAUATWW
:59:723491524
C. KLEIN
BLOEMENGRACHT 15
AMSTERDAM
:71A:SHA
:71F:USD10,
:71F:USD10,
:72:/INS/CHASUS33
-}{5:{MAC:75D138E4}{CHK:DE1B0D71FA96}}

```

**Comment by danielquinn** [comment]
Wow, thank you for being so thorough.  I've never needed to make use of these fields so I never noticed that they were missing in some cases!  I'll re-open and fix this.

In the mean time, you don't happen to have an example of a message with a `13C:` value in it do you?

**Comment by danielquinn** [comment]
Ok!  It turns out that the problem was my source spec!  I referenced [this Wikipedia page](https://en.wikipedia.org/wiki/MT103) while I was writing the parser, and it excludes parts of the format which, when not factored into the parser, meant we didn't get everything we should.

I've updated `master` and released `1.1.0` to PyPI.  Please let me know if it works for you!

**Comment by bbroto06** [comment]
I'm extremely happy. All thanks to you. Its working now.
